### ISSUE TITLE
Add support to inline templates(inside ts/js files)

### ISSUE THREAD

**Issue by flashios09** [ember-template-lint/ember-template-lint#886]
Hi,
The unstable ember language server(https://github.com/flashios09/ember-language-server/commit/7d3cbc33a4b60b7bf602aad8cfe0b8d316e95178) now supports the linting for inline templates(inside ts/js).
I think that we can move it to the `ember-template-lint`, just extract the inline templates before:
```js
verify(options) {
    //...
    const source = (getExtension(options.moduleId) === '.hbs') ? 
      options.source : 
      searchAndExtractHbs(options.source);
    try {
      precompile(stripBom(source), {
        moduleName: options.moduleId,
        rawSource: stripBom(source),
        plugins: {
          ast: this.buildASTPlugins(pluginConfig),
        },
      });
    } catch (error) {
      let message = {
        fatal: true,
        moduleId: options.moduleId,
        message: error.message,
        source: error.stack,
        severity: 2,
      };
    }
    // ...
  }
```
like this:
- we don't need to use any extra plugin or another linter(eslint ...).
- we don't need to duplicate the rules, **one unified config file** will be used for hbs, js and ts.
- the **ember-cli-template-lint** will also support the inline templates.

**Comment by rwjblue** [comment]
I think it would take a few more changes, but seems _generally_ good to me. 

Some things that this will have to address when implemented:

* The bin script globing would have to change (it defaults to `**/*.hbs`)
* Ensure that the reported line and column numbers in all of the reported failures offsets are correct to what is actually in the file
* We’d need to figure out how to parse the JS/TS files (possibly requiring “parser” config like eslint does) so that we don’t have issues with “custom syntax” (e.g. decorators)

**Comment by flashios09** [comment]
> The bin script globing would have to change (it defaults to **/*.hbs).

I know about this and i think it can be updated to support inline template too. Just remove/update the *extension* check.

> Ensure that the reported line and column numbers in all of the reported failures offsets are correct to what is actually in the file

The `searchAndExtractHbs` function works fine with indentation(column, line), you can check the tests here https://github.com/flashios09/extract-tagged-template-literals/blob/master/src/__tests__/searchAndExtract.test.ts

> We’d need to figure out how to parse the JS/TS files (possibly requiring “parser” config like eslint does) so that we don’t have issues with “custom syntax” (e.g. decorators)

Sorry i didn't get this point, can you explain more please ?

**Comment by gabrielcsapo** [comment]
Any update on this? This would be super great to have added.

**Comment by NullVoxPopuli** [comment]
couldn't we get around the parser issue by ignoring it all together, and pretty much just treat the files as text?

for example:
 step 1: regexily extract imports
 step 2: actually parse the imports to look at what `hbs` is called (or skip step 1 entirely and assume hbs (this is the better option for support markdown)
 step 3: find all occurances of `hbs` or whatever-it-was-renamed to (via readLine or similar iteration, then you have your line offset).
 step 4: extract the surrounding code via simple in-plugin parser (to find matching backtick (and this is safeish, because we don't allow nested backticks, I think?))
 step 5: send content between backticks to ember-template-lint with line/column offsets
 step 6: ???
 step 7: profit

**Comment by lifeart** [comment]
@NullVoxPopuli this is how linting works in UELS

**Comment by NullVoxPopuli** [comment]
right, but I need linting `hbs` sections (of tests and inline usage outside of tests) to happen in C.I. as well (unless it's possible to use UELS instead of ember-template-lint for linting in C.I.?)

**Comment by lifeart** [comment]
@NullVoxPopuli it’s possible, I have reference plugin for it - https://github.com/lifeart/ember-template-lint-typed-template you have to call linter from it instead of typed-templates

**Comment by lifeart** [comment]
Woo, here is long time, now all this cases supported natively, thank you to all contributors!
